### ISSUE TITLE
[YAML] Regression in parsing inline objects since version 4.4

### ISSUE THREAD

**Issue by cebe** [symfony/symfony#34805]
**Symfony version(s) affected**: 4.4.x (it works in 3.x as well as 4.x up to 4.3, [4.4 or greater is failing](https://travis-ci.org/cebe/php-openapi/builds/620519083))

**Description**  

Parsing inline Objects in YAML is broken since 4.4, see example script below.

**How to reproduce**  

```php
<?php
require __DIR__ . '/vendor/autoload.php';

$yaml = \Symfony\Component\Yaml\Yaml::parse(<<<YAML
data:
  example:
    {
      "total": 2,
      "apis": [
        {
          "apiKey": "oa_citations",
          "apiVersionNumber": "v1",
        },
        {
          "apiKey": "cancer_moonshot",
          "apiVersionNumber": "v1",
        }
      ]
    }
YAML
);

print_r($yaml);
```

On yaml versions <4.4, it shows the correct parsing:

```
Array
(
    [data] => Array
        (
            [example] => Array
                (
                    [total] => 2
                    [apis] => Array
                        (
                            [0] => Array
                                (
                                    [apiKey] => oa_citations
                                    [apiVersionNumber] => v1
                                )
                            [1] => Array
                                (
                                    [apiKey] => cancer_moonshot
                                    [apiVersionNumber] => v1
                                )
                        )
                )
        )
)
```

since 4.4 I get the following error:

```php
Fatal error: Uncaught Symfony\Component\Yaml\Exception\ParseException: Malformed inline YAML string: {
  "total": 2,
  "apis": [
    {
      "apiKey": "oa_citations",
      "apiVersionNumber": "v1",
    },
    {
      "apiKey": "cancer_moonshot",
      "apiVersionNumber": "v1",
    }
  ]
} at line 15 (near "}"). in /home/cebe/dev/layer5/layer5-ng/vendor-dev/cebe/php-openapi/vendor/symfony/yaml/Inline.php on line 300

Symfony\Component\Yaml\Exception\ParseException: Malformed inline YAML string: {
  "total": 2,
  "apis": [
    {
      "apiKey": "oa_citations",
      "apiVersionNumber": "v1",
    },
    {
      "apiKey": "cancer_moonshot",
      "apiVersionNumber": "v1",
    }
  ]
} at line 15 (near "}"). in /home/cebe/dev/layer5/layer5-ng/vendor-dev/cebe/php-openapi/vendor/symfony/yaml/Inline.php on line 300
```

**Additional info**

could be related to https://github.com/symfony/yaml/commit/110e3e9eaa7827def84cb1cfee95a6cfde4bf608 (https://github.com/symfony/symfony/pull/33658)

**Comment by cebe** [comment]
Just came accross another example that fails in 4.3 as well as 4.4, so no regression but as far as I see very related:

```php
<?php
require __DIR__ . '/vendor/autoload.php';

$yaml = \Symfony\Component\Yaml\Yaml::parse(<<<YAML
data:
  example: {
    "versions": [
        1,
        2
    ]
  }
YAML
);
print_r($yaml);
```

**Comment by xabbuh** [comment]
The inline object notation spanning multiple lines did only work by chance for some cases in 4.3. Though in 4.4 we explicitly added support for it so that's the place where we need to fix it. I am looking into it.

**Comment by xabbuh** [comment]
This will finally be fixed by #33763.
